### PR TITLE
Issue #3655 simplifications of Cookie handling

### PR DESCRIPTION
**Pull request by joakime**
#3655 replaces #3658

### CODE REVIEW

File: jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java [hunk 1]
```diff
@@ -161,4 +189,200 @@ public String asString()
...
+        if (_comment != null)
+        {
+            buf.append(";Comment=");
+            quoteOnlyOrAppend(buf,_comment,isQuoteNeededForCookie(_comment));
+        }
+        return buf.toString();
+    }
+
+    public String getRFC6265SetCookie()
+    {
```
Comment (janbartel) [inline jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java]: Is there some more refactoring that could be done between the getRFC2965SetCookie and getRFC6265SetCookie methods? The method bodies look almost (but not quite) identical.

File: jetty-server/src/main/java/org/eclipse/jetty/server/Response.java [hunk 2]
```diff
@@ -170,30 +165,13 @@ public void included()
...
-                cookie.getPath(),
-                cookie.getMaxAge(),
-                cookie.isSecure(),
-                cookie.isHttpOnly());
+
+        // add the set cookie
+        _fields.add(new SetCookieHttpField(cookie, getHttpChannel().getHttpConfiguration().getResponseCookieCompliance()));
+
+        // Expire responses with set-cookie headers so they do not get cached.
+        _fields.put(__EXPIRES_01JAN1970);
```
Comment (janbartel) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Response.java]: Is this line new? Did the old code put in the default expiry of 01JAN1970?

File: jetty-http/src/test/java/org/eclipse/jetty/http/HttpCookieTest.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        
+        httpCookie = new HttpCookie("name", "value%=", null, null, -1, false, false, null, 0);
+        setCookie=httpCookie.getRFC2965SetCookie();
+        assertEquals("name=value%=",setCookie);
+    }
+
+    @Test
+    public void testSetRFC6265Cookie() throws Exception
+    {
+        //  HttpCookie(String name, String value, String domain, String path, long maxAge, boolean httpOnly, boolean secure, String comment, int version)
```
Comment (janbartel) [inline jetty-http/src/test/java/org/eclipse/jetty/http/HttpCookieTest.java]: If these lines aren't needed, delete.

File: jetty-server/src/main/java/org/eclipse/jetty/server/Response.java (hunk 2 above)
Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/Response.java]: That line was present in 9.4.18.v20190429 too.

See: 
 * https://github.com/eclipse/jetty.project/blob/jetty-9.4.18.v20190429/jetty-server/src/main/java/org/eclipse/jetty/server/Response.java#L342-L359
 * https://github.com/eclipse/jetty.project/blob/jetty-9.4.18.v20190429/jetty-server/src/main/java/org/eclipse/jetty/server/Response.java#L422-L440

File: jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java (hunk 1 above)
Comment (joakime) [reply jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java]: mandatory spaces, lack of quoting, and different syntax checks are the biggest differences I can see.

Comment (gregw) [reply jetty-http/src/main/java/org/eclipse/jetty/http/HttpCookie.java]: I had a look and there is would be too difficult and error prone to merge these.

**Pull request merged**